kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -320,6 +320,26 @@ std::optional<Updates> KeyValueBlockchain::getBlockUpdates(BlockId block_id) con
   return Updates{std::move(raw->data.updates)};
 }
 
+std::unordered_map<std::string, std::vector<std::string>> KeyValueBlockchain::getBlockStaleKeys(BlockId block_id) {
+  // Get block node from storage
+  auto block = block_chain_.getBlock(block_id);
+  if (!block) {
+    const auto msg = "Failed to get block node for block ID = " + std::to_string(block_id);
+    throw std::runtime_error{msg};
+  }
+
+  std::unordered_map<std::string, std::vector<std::string>> stale_keys;
+  // Iterate over groups and call corresponding deleteGenesisBlock,
+  // Each group is responsible to put its deletes into the batch
```

Comment:
I think this comment is not relevant here.

Reply:
fixed